doc/vim-go.txt

```diff
@@ -211,11 +211,67 @@ COMMANDS                                                          *go-commands*
     given, it will jump to the declaration under the cursor. By default the
-    mapping `gd` is enabled to invoke GoDef for the identifier under the cursor.
-    See |g:go_def_mapping_enabled| to disable it.
+    CTRL-] key and the mapping `gd` are enabled to invoke :GoDef for the
+    identifier under the cursor. See |g:go_def_mapping_enabled| to disable them.
+
+    vim-go also keeps a per-window location stack, roughly analagous to how
+    vim's internal |tags| functionality works. This is pushed to every time a
+    jump is made using the GoDef functionality. In essence, this is a LIFO list
+    of file locations you have visited with :GoDef that is retained to help you
+    navigate software.
+
+                                                                *:GoDefStack*
+:GoDefStack
+
+    This command displays the entries in the GoDef stack.  Its output looks
+    like this:
+
+      1 /path/to/first/file.go|1187 col 16|AddThing func(t *Thing)
+    > 2 /path/to/thing/thing.go|624 col 19|String() string
+      3 /path/to/thing/thing.go|744 col 6|func Sprintln(a ...interface{}) string
+
+    This list shows the identifiers that you jumped to and the file and cursor
+    position before that jump.  The older jumps are at the top, the newer at the
+    bottom.
+
+    The '>' points to the active entry.  This entry and any newer entries below
+    it will be replaced if a jump is done from this location. The CTRL-t and
+    |:GoDefPop| command will use the position above the active entry.
+
+                                                                  *:GoDefPop*
+:GoDefPop [count]
+CTRL-t
+
+    Navigate to the [count] earlier entry in the jump stack, retaining the newer
+    entries. If no argument is given, it will jump to the next most recent entry
+    (`:GoDefPop 1`).  If [count] is greater than the number of prior entries,
+    an error will be printed and no jump will be performed.
+
+    If you have used :GoDefPop to jump to an earlier location, and you issue
+    another :GoDef command, the current entry will be replaced, and all newer
+    entries will be removed, effectively resuming the stack at that location.
+
+    By default [count]CTRL-t is enabled to invoke :GoDefPop.  Similarly, hitting
+    CTRL-t without a prior count is equivalent to `:GoDefPop 1`.  See
+    |g:go_def_mapping_enabled| to disable this.
+
+                                                                 *:GoDefJump*
+:GoDefJump [number]
+
+    Jumps to a given location in the jump stack, retaining all other entries.
+    If no argument is given, it will print out an interactive list of all items
+    in the stack, from which you can select one, if you choose.
+
+    For details on the format of this list, see |:GoDefStack|.
+
+    Jumps to non-existent entries will print an informative message, but are
+    otherwise a noop.  As with |:GoDefPop|, any :GoDef command after a jump
+    will replace the current entry and remove all newer entries.
```

Comment:
I agree.  They can always `:q`, `q`, `<ESC>` or select the current entry.  What about the cursor? I would like to turn it off so that it is more like a menu. Thoughts on that?